freeipa/src/main/java/com/sequenceiq/freeipa/service/config/KerberosConfigUpdateService.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.freeipa.service.config;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.kerberos.KerberosConfig;
+import com.sequenceiq.freeipa.kerberos.KerberosConfigService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class KerberosConfigUpdateService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KerberosConfigUpdateService.class);
+
+    @Inject
+    private KerberosConfigService kerberosConfigService;
+
+    @Inject
+    private StackService stackService;
+
+    public KerberosConfig updateNameservers(Long stackId) {
+        Stack stack = getStackWithInstanceMetadata(stackId);
+        String environmentCrn = stack.getEnvironmentCrn();
+        KerberosConfig kerberosConfig = kerberosConfigService.get(environmentCrn);
```

Comment:
this won't return all. you have to use `kerberosConfigRepository.findByAccountIdAndEnvironmentCrnAndArchivedIsFalse` on repository level, and this is not exposed on service level right now